LeapEngine/Graphics/GraphicsObject.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+namespace leap
+{
+	template<typename T>
+	class GraphicsObject final
```

Comment:
Wrapper that has the same functionality as a shared pointer? Make this a general class and place it in Utils.